Tests/LibJS/test-test262.cpp

```diff
@@ -142,16 +143,19 @@ class Test262RunnerHandler {
         close(write_pipe_fds[0]);
         close(read_pipe_fds[1]);
 
-        auto infile = Core::File::construct();
-        infile->open(read_pipe_fds[0], Core::OpenMode::ReadOnly, Core::File::ShouldCloseFileDescriptor::Yes);
+        auto infile_or_error = Core::Stream::File::adopt_fd(read_pipe_fds[0], Core::Stream::OpenMode::Read);
 
-        auto outfile = Core::File::construct();
-        outfile->open(write_pipe_fds[1], Core::OpenMode::WriteOnly, Core::File::ShouldCloseFileDescriptor::Yes);
+        auto outfile_or_error = Core::Stream::File::adopt_fd(write_pipe_fds[1], Core::Stream::OpenMode::Write);
 
-        return make<Test262RunnerHandler>(pid, move(infile), move(outfile));
+        if (infile_or_error.is_error() || outfile_or_error.is_error()) {
+            kill(pid, SIGKILL);
+            return nullptr;
```

Comment:
Hmm, this seems weirdly factored. Could we do something like:
```cpp
ArmedScopeGuard runner_killer { [pid]() { kill(pid, SIGKILL); } };

// ...

auto infile = TRY(Core::Stream::File::adopt_fd(...));
auto outfile = TRY(Core::Stream::File::adopt_fd(...));

runner_killer.disarm();
return make<Test262RunnerHandler>(...);
```

Reply:
Would infile be closed if the adopting of outfile failed? (I know that isn't handled currently either)

Reply:
Of course, through the power of RAII. :^)